private/rules/coursier.bzl

```diff
@@ -925,12 +925,35 @@ def make_coursier_dep_tree(
     if (exec_result.return_code != 0):
         fail("Error while fetching artifact with coursier: " + exec_result.stderr)
 
-    return deduplicate_and_sort_artifacts(
+    dep_tree = deduplicate_and_sort_artifacts(
         json.decode(repository_ctx.read(repository_ctx.path("dep-tree.json"))),
         artifacts,
         excluded_artifacts,
         _is_verbose(repository_ctx),
     )
+    return rewrite_files_attribute_if_necessary(repository_ctx, dep_tree)
+
+def rewrite_files_attribute_if_necessary(repository_ctx, dep_tree):
+    # There are cases where `coursier` will download both the pom and the
+    # jar but will include the path to the pom instead of the jar in the
+    # `file` attribute. This differs from both gradle and maven. Massage the
+    # `file` attributes if necessary.
+    # https://github.com/bazelbuild/rules_jvm_external/issues/1250
+    amended_deps = []
+    for dep in dep_tree["dependencies"]:
+        if not dep.get("file", None):
+            amended_deps.append(dep)
+            continue
+
+        jar_path = dep["file"].removesuffix(".pom") + ".jar"
```

Comment:
This could be https://github.com/bazelbuild/bazel-skylib/blob/main/docs/paths_doc.md#pathsreplace_extension, and guarded with a `endswith(".pom")`.

Reply:
Done